### PR TITLE
Fix bpo-30526: make TextIOWrapper.{line_buffering, write_through} writable

### PR DESCRIPTION
**Pull request by pitrou**
This simplifies the reconfiguration of standard streams as it can now be done in-place.

### CODE REVIEW

**Comment by pitrou** [comment]
The AppVeyor failure is unrelated.

File: Doc/library/io.rst [hunk 1]
```diff
@@ -908,6 +908,16 @@ Text I/O
 
       Whether line buffering is enabled.
```
Comment (benjaminp) [inline Doc/library/io.rst:909]: should also say that writing to it flushes the stream

File: Doc/library/io.rst [hunk 2]
```diff
@@ -908,6 +908,16 @@ Text I/O
...
 
+      .. versionchanged:: 3.7
+         This argument is now writable.
+
+   .. attribute:: write_through
+
+      Whether writes are passed immediately to the underlying binary
+      buffer.  This argument is writable.
```
Comment (benjaminp) [inline Doc/library/io.rst:917]: ditto

**Review by benjaminp — COMMENTED** [review]
so what happens when I set `line_buffering = True` and `write_through = True`?

**Comment by pitrou** [comment]
https://github.com/python/cpython/pull/1922 is now preferred.

**Comment by pitrou** [comment]
> so what happens when I set line_buffering = True and write_through = True?

Same as before: `write_through` takes precedence (everything is written through).

**Comment by pitrou** [comment]
Superseded by #1922.

**Pull request closed without merging**